libfreerdp/codec/clear.c

```diff
@@ -1007,8 +1036,7 @@ INT32 clear_decompress(CLEAR_CONTEXT* clear, const BYTE* pSrcData, UINT32 SrcSiz
 
 	if (glyphFlags & CLEARCODEC_FLAG_CACHE_RESET)
 	{
-		clear->VBarStorageCursor = 0;
-		clear->ShortVBarStorageCursor = 0;
+		clear_reset_vbar_storage(clear);
```

Comment:
@akallabeth agreed, this should not be merged unless we can hit this particular code path and ensure it doesn't break

Reply:
to avoid causing another possible regression like with the xcrush context reset, I will modify the current PR to keep the current reset behaviour, just with cleaned up code. I'll add a bool parameter to do a "full" internal reset, and make sure this call site only does a "shallow" reset that only touches the indices. I know there was an issue years ago with pcap replaying, but it would be safer to revisit it at the very end.